lib/acme/certificate.rb

```diff
@@ -9,6 +9,39 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
+
+    cert = OpenSSL::X509::Certificate.new
+    cert.subject = cert.issuer = OpenSSL::X509::Name.parse "/CN=#{subjectAltDomains.first}"
```

Comment:
This seems like a workaround in order to avoid using the proper API for this, `OpenSSL::X509::Name.new([["CN", hostname, OpenSSL::ASN1::UTF8STRING]])`